tests/integration/courseSlots.test.js

```diff
@@ -141,478 +141,477 @@ describe('COURSE SLOTS ROUTES - POST /courseslots', () => {
+      };
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/courseslots/${courseSlotsList[10]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload,
+      });
+
+      expect(response.statusCode).toBe(403);
+    });
+  });
+
+  describe('COACH', () => {
+    beforeEach(async () => {
+      authToken = await getToken('coach');
+    });
+
+    it('should update course slot (intra)', async () => {
+      const payload = { startDate: '2020-03-04T09:00:00.000Z', endDate: '2020-03-04T11:00:00.000Z' };
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/courseslots/${courseSlotsList[0]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload,
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    it('should update course slot (intra_holding)', async () => {
+      const payload = { startDate: '2020-03-04T09:00:00.000Z', endDate: '2020-03-04T11:00:00.000Z' };
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/courseslots/${courseSlotsList[10]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload,
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    it('should return 403 if user from other company try to update course slot (intra)', async () => {
+      const payload = { startDate: '2020-03-04T09:00:00.000Z', endDate: '2020-03-04T11:00:00.000Z' };
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/courseslots/${courseSlotsList[0]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload,
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    it('should return 403 if user try to update slot from course with good holding but without company (intra_holding)',
+      async () => {
+        const payload = { startDate: '2020-03-04T09:00:00.000Z', endDate: '2020-03-04T11:00:00.000Z' };
+        const response = await app.inject({
+          method: 'PUT',
+          url: `/courseslots/${courseSlotsList[11]._id}`,
+          headers: { Cookie: `alenvi_token=${authToken}` },
+          payload,
+        });
+
+        expect(response.statusCode).toBe(403);
+      });
+  });
+
+  describe('Other roles', () => {
+    it('should a 200 as user is course trainer', async () => {
+      authToken = await getTokenByCredentials(trainer.local);
+      const payload = { startDate: '2020-03-04T09:00:00.000Z', endDate: '2020-03-04T11:00:00.000Z' };
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/courseslots/${courseSlotsList[2]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload,
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    const roles = [
+      { name: 'helper', expectedCode: 403 },
+      { name: 'planning_referent', expectedCode: 403 },
+      { name: 'trainer', expectedCode: 403 },
```

Comment:
Ici je trouve que c'est pas hyper clair car quand tu regardes dans les seeds, trainer est formateur de la formation coursesList[3] mais en fiat on ne parle pas du meme utilisateur. 

Je supprimerai ce test pour ajouter une section `describe('TRAINER', () => {}`  juste au dessus.

Dans cette section , je déplacerai le tests "should return 200 as user is course trainer" et j'ajouterai un test "should return 403 as user is trainer but not course trainer"

Pour le premier test, tu peux te connecter avec le token `trainer.local` en utilisant getTokenByCredentials et pour le deuxieme tu peux utliiser le role `trainer` qui correspond au `trainer` defini dans `usersList` du fichier `AuthUsersSeed`

Reply:
je n'arrive pas a récupérer trainer de authUserSeed parce que le mot trainer est déjà importer depuis courseSlotSeed, je ne comprends pas comment faire